optimum/neuron/trainers.py

```diff
@@ -421,14 +421,17 @@ def _save_xla(self, output_dir: Optional[str] = None):
             self.tokenizer.save_pretrained(output_dir)
 
     def save_model(self, output_dir: Optional[str] = None, _internal_call: bool = False):
-        if output_dir is None:
-            output_dir = self.args.output_dir
+        if not os.environ.get("NEURON_PARALLEL_COMPILE"):  # Avoid unnecessary model saving during precompilation
+            if output_dir is None:
+                output_dir = self.args.output_dir
 
-        self._save_xla(output_dir)
+            self._save_xla(output_dir)
 
-        # Push to the Hub when `save_model` is called by the user.
-        if self.args.push_to_hub and not _internal_call:
-            self.push_to_hub(commit_message="Model save")
+            # Push to the Hub when `save_model` is called by the user.
+            if self.args.push_to_hub and not _internal_call:
+                self.push_to_hub(commit_message="Model save")
+        else:
+            logger.info("Skipping trainer.save_model() while running under neuron_parallel_compile")
```

Comment:
I think there is a very quick incidental compilation related to checkpointing - I will time it to see if it's significant and update here.

This is really meant to avoid the several minutes waiting and 38GB storage used up by checkpointing during neuron_parallel_compile. We do something similar for [neuronx-nemo-megatron](https://github.com/aws-neuron/neuronx-nemo-megatron/blob/main/nemo/examples/nlp/language_modeling/train_setup.sh#L111)

Reply:
Alright, then let's do it!